API.md

```diff
@@ -73,7 +73,8 @@ Custom types may be registered via
 ```
 openmct.types.addType('my-type', new openmct.Type({
     label: "My Type",
-    description: "This is a type that I added!"
+    description: "This is a type that I added!",
+    creatable: true
 });
```

Comment:
Only reason we had the constructor in there was to be "less declarative" which sure is a silly reason. Pushed the recommended change & updated corresponding documentation